### PR TITLE
Adds a Diagnostic Welding HUD for roboticists

### PR DESCRIPTION
**Pull request by etherware-novice**
# About The Pull Request
Adds a new hud type, being a combination of a diagnostic hud and welding shield. The welding shield part can be "pushed down" to quickly weld off items.

## Why It's Good For The Game
Roboticists currently have to use both the welding shield and diagnostic hud frequently to do their job well, which can be somewhat annoying if you want or need to use the head slot for other gear. This pr would give Roboticists a new option for optimal productivity.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

diagweld_floor

diagweld_research

diagweld_work

</details>

## Changelog

:cl:
add: Added Diagnostic Welding HUD
:tweak: Added Diag-Weld HUD to the "Night Vision HUD" research node
:tweak: Added Diag-Weld HUD to the RoboDrobe as a premium item
/:cl:

### CODE REVIEW

**Comment by nednaZ** [comment]
You need to update your Master branch. This PR modifies catgloves.

**Comment by nednaZ** [comment]
You need to update your Master branch. This PR modifies catgloves.

**Comment by etherware-novice** [comment]
> You need to update your Master branch. This PR modifies catgloves.

how do i do that?

File: code/modules/research/designs/misc_designs.dm [hunk 1]
```diff
@@ -53,6 +53,16 @@
 	category = list("Equipment")
 	departmental_flags = DEPARTMENTAL_FLAG_SCIENCE
 
+/datum/design/diagnostic_hud_weld
```
Comment (nednaZ) [inline code/modules/research/designs/misc_designs.dm:56]: Needs to go into the monkestation modular folder

File: code/modules/research/techweb/all_nodes.dm [hunk 2]
```diff
@@ -617,7 +617,7 @@
 	display_name = "Night Vision Technology"
 	description = "Allows seeing in the dark without actual light!"
 	prereq_ids = list("integrated_HUDs", "adv_engi", "emp_adv")
-	design_ids = list("health_hud_night", "security_hud_night", "diagnostic_hud_night", "night_visision_goggles", "nvgmesons")
+	design_ids = list("health_hud_night", "security_hud_night", "diagnostic_hud_night", "diagnostic_hud_weld", "night_visision_goggles", "nvgmesons")
```
Comment (nednaZ) [inline code/modules/research/techweb/all_nodes.dm:620]: Night vision tech seems like an odd place for it.

Also, put a comment at the end that there is a monkestation edit.

File: code/modules/vending/wardrobes.dm [hunk 3]
```diff
@@ -186,6 +186,7 @@
 					/obj/item/clothing/under/costume/mech_suit = 2,
 					/obj/item/clothing/under/costume/mech_suit/white = 2,
 					/obj/item/clothing/under/costume/mech_suit/blue = 2,)
+	premium = list ( /obj/item/clothing/glasses/hud/diagnostic/weld = 1 )
```
Comment (nednaZ) [inline code/modules/vending/wardrobes.dm:189]: No, this is a tech item and should be produced only

**Comment by etherware-novice** [comment]
> I'm unsure about the balance of this one.
> 
> Glasses that can be printed and provide flash protection with zero penalty may not work.
> 
> 
> 
> Icons needs to be in the monkestation module folders

it only provides flash protection when down, similar to standard welding equipment. it also requires quite a bit of research to unlock (night vision hud)

File: code/modules/clothing/glasses/hud.dm [hunk 4]
```diff
@@ -101,6 +101,19 @@
...
+	desc = "A diagnostic HUD fitted with a small shield for welding purposes. Useful for optimal cyborg repair."
+	icon_state = "diagnostichudweld"
+	actions_types = list(/datum/action/item_action/toggle)
+	flash_protect = 2
+	tint = 2
+	visor_vars_to_toggle = VISOR_FLASHPROTECT | VISOR_TINT
+	flags_cover = GLASSESCOVERSEYES
+
+/obj/item/clothing/glasses/hud/diagnostic/weld/attack_self(mob/user)
+	weldingvisortoggle(user)
```
[the remark this answers is not in the document]
Comment (dwasint) [reply code/modules/clothing/glasses/hud.dm]: weldingvisortoggle is defined in /obj/item/clothing/glasses? and the actual proc is in /obj/item/clothing

File: code/modules/clothing/glasses/hud.dm [hunk 5]
```diff
@@ -101,6 +101,19 @@
 	hud_type = DATA_HUD_DIAGNOSTIC_BASIC
 	glass_colour_type = /datum/client_colour/glass_colour/lightorange
 
+/obj/item/clothing/glasses/hud/diagnostic/weld
+	name = "diagnostic welding HUD"
+	desc = "A diagnostic HUD fitted with a small shield for welding purposes. Useful for optimal cyborg repair."
+	icon_state = "diagnostichudweld"
+	actions_types = list(/datum/action/item_action/toggle)
+	flash_protect = 2
```
[the remark this answers is not in the document]
Comment (dwasint) [reply code/modules/clothing/glasses/hud.dm]: weldingvisortoggle invert that

**Comment by Serxule** [comment]
i mean the diagnostic hud implant already exists, combined with welding shield eyes, and they could also just use a welding helmet (which most do since goggles arent common)

File: code/modules/clothing/glasses/hud.dm [hunk 6]
```diff
@@ -101,6 +101,7 @@
 	hud_type = DATA_HUD_DIAGNOSTIC_BASIC
 	glass_colour_type = /datum/client_colour/glass_colour/lightorange
 
+
```
Comment (Zonespace27) [inline code/modules/clothing/glasses/hud.dm:104]: why whitespace

File: code/modules/research/techweb/all_nodes.dm [hunk 7]
```diff
@@ -617,7 +617,7 @@
 	display_name = "Night Vision Technology"
 	description = "Allows seeing in the dark without actual light!"
 	prereq_ids = list("integrated_HUDs", "adv_engi", "emp_adv")
-	design_ids = list("health_hud_night", "security_hud_night", "diagnostic_hud_night", "night_visision_goggles", "nvgmesons")
+	design_ids = list("health_hud_night", "security_hud_night", "diagnostic_hud_night", "diagnostic_hud_weld", "night_visision_goggles", "nvgmesons")  // monkestation edit
```
Comment (Zonespace27) [inline code/modules/research/techweb/all_nodes.dm:620]: this has nothing to do with nightvision, put it elsewhere

File: monkestation/code/modules/research/designs/misc_designs.dm [hunk 8]
```diff
@@ -7,3 +7,14 @@
 	build_path = /obj/item/clothing/gloves/cargo_gauntlet
 	category = list("Equipment")
 	departmental_flags = DEPARTMENTAL_FLAG_CARGO
+
+
```
Comment (Zonespace27) [inline monkestation/code/modules/research/designs/misc_designs.dm:11]:
```suggestion
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/research/techweb/all_nodes.dm [hunk 9]
```diff
@@ -701,7 +701,7 @@
 	display_name = "Cybernetic Implants"
 	description = "Electronic implants that improve humans."
 	prereq_ids = list("adv_biotech", "datatheory")
-	design_ids = list("ci-nutriment", "ci-breather", "ci-gloweyes", "ci-welding", "ci-medhud", "ci-sechud", "ci-diaghud", "ci-lighter") // MONKE EDIT - added ci-lighter
+	design_ids = list("ci-nutriment", "ci-breather", "ci-gloweyes", "ci-welding", "ci-medhud", "ci-sechud", "ci-diaghud", "ci-lighter", "diagnostic_hud_weld") // MONKE EDIT - added ci-lighter (and diag hud)
```
[the remark this answers is not in the document]
Comment (etherware-novice) [reply code/modules/research/techweb/all_nodes.dm:704]: which section would you like me to put it in?

[the remark this answers is not in the document]
Comment (nednaZ) [reply code/modules/research/techweb/all_nodes.dm:704]: My suggestion would be "Experimental Tools"

**Review by nednaZ — CHANGES_REQUESTED** [review]
No worn icon, needs some indication that they are flashproof.

**Review by nednaZ — APPROVED** [review]
Looks good to me.

**Pull request merged**